pkg/controller/statefulset/stateful_set.go

```diff
@@ -95,8 +97,9 @@ func NewStatefulSetController(
 				setInformer.Lister(),
 				podInformer.Lister(),
 				pvcInformer.Lister(),
-				recorder,
-			),
+				recorder),
+			NewRealStatefulSetStatusUpdater(kubeClient, setInformer.Lister()),
+			history.NewHistory(kubeClient, revInformer.Lister()),
 		),
 		pvcListerSynced: pvcInformer.Informer().HasSynced,
```

Comment:
Since you are using the controllerRevision cache to list CRs, you should wait for it to be synced before starting the sts controller. Means you need a crListerSynced field which you then pass in controller.WaitForCacheSync (L150).